tests/unit/common/test_bootstrap_macos.py

```
@@ -26,193 +24,121 @@ def setup_bootstrap_mocks(
+            assert mock_dup2.call_args_list == [
+                call(99, 0),
+                call(99, 1),
+                call(99, 2),
+            ]
+            mock_close.assert_called_once_with(99)
+
+    def test_redirect_creates_python_streams_from_fds(self):
+        """sys.stdin/stdout/stderr are recreated from OS-level FDs."""
+        mock_streams = {0: MagicMock(), 1: MagicMock(), 2: MagicMock()}
+
+        def fdopen_side_effect(fd, mode="r"):
```

Ruff (0.15.2)

[warning] Unused function argument: `mode`

(ARG001)